src/Board.py

```diff
@@ -90,21 +90,23 @@ def undoLastMove(self) -> None:
             rook = lastMove.specialMovePiece
 
             self.movePieceToPosition(king, lastMove.oldPos)
-            self.movePieceToPosition(rook, lastMove.rookMove.oldPos)
-
             king.movesMade -= 1
-            rook.movesMade -= 1
+
+            if rook:
+                self.movePieceToPosition(rook, lastMove.rookMove.oldPos)
+                rook.movesMade -= 1
```

Comment:
Recent changes triggered further Mypy errors, here it doesn't like that `specialMovePiece` by default is `None` and `None` can neither be an input to `movePieceToPosition()` nor it has `movesMade`. Same happened to `pawnTaken` few lines later.

Reply:
For me running `mypy --show-error-codes --pretty --strict src/` on this branch in the main directory `mypy` reports:
```bash
Success: no issues found in 15 source files
```

Reply:
Yes, because I've added this two `if`s, and I just mentioned it here so you know why further code changes took place. Without them you'd have:
```
~/Projects/command-line-chess (add-mypy ✗) mypy --show-error-codes --pretty --strict .
src/Board.py:96: error: Argument 1 to "movePieceToPosition" of "Board" has incompatible type "None"; expected "Piece"  [arg-type]
                self.movePieceToPosition(rook, lastMove.rookMove.oldPos)
                                         ^~~~
src/Board.py:96: error: "None" has no attribute "oldPos"  [attr-defined]
                self.movePieceToPosition(rook, lastMove.rookMove.oldPos)
                                               ^~~~~~~~~~~~~~~~~~~~~~~~
src/Board.py:97: error: "None" has no attribute "movesMade"  [attr-defined]
                rook.movesMade -= 1
                ^~~~~~~~~~~~~~
src/Board.py:103: error: Argument 1 to "append" of "list" has incompatible type "Optional[Piece]"; expected "Piece"  [arg-type]
                self.pieces.append(pawnTaken)
                                   ^~~~~~~~~
src/Board.py:104: error: Item "None" of "Optional[Piece]" has no attribute "side"  [union-attr]
                if pawnTaken.side == WHITE:
                   ^~~~~~~~~~~~~~
src/Board.py:106: error: Item "None" of "Optional[Piece]" has no attribute "side"  [union-attr]
                if pawnTaken.side == BLACK:
                   ^~~~~~~~~~~~~~
Found 6 errors in 1 file (checked 16 source files)

```

Reply:
Anyways I'm happy with this PR now, I also have another one ready for making flake8 happy but will push it once this is merged.